packages/react-server/src/ReactFlightServer.js

```diff
@@ -390,18 +457,29 @@ export function resolveModelToJSON(
   switch (value) {
     case REACT_ELEMENT_TYPE:
       return '$';
+    case REACT_PROVIDER_TYPE:
+      return '!';
```

Comment:
If you introduce new special codes, they need to be escaped in escapeStringValue. Otherwise user strings would also trigger this case. Which would likely not be that uncommon. We only have so many characters we can reasonably special case to always escape and therefore make those strings longer.

I don't think this is common enough of a type to warrant that.

It should work to just let it pass through. It'll encode it as a symbol chunk and then just refer to that symbol by ID. So you can just remove these parts.